### PR TITLE
Feature/frontend ci

### PR DESCRIPTION
**Pull request by gmikio**
## Feature/frontend-ci: Fix Build & Lint Issues

This pull request aims to merge the `feature/frontend-ci` branch into `staging`. The main focus of this update was to address several build and linting issues, ensuring that the project can now successfully run through all CI/CD checks.

### Key Changes
- **Fixed ESLint Errors and Warnings**: Resolved all linting issues to ensure the codebase meets quality standards.
- **Resolved TypeScript Build Issues**: Addressed TypeScript errors that were preventing successful builds.
- **Updated Docker Configuration**: Modified the Dockerfile and related configurations to enable successful Docker builds for deployment.

These improvements are crucial for preparing the frontend for a smooth Docker build process and subsequent deployment to production environments. Please review and approve to integrate these important fixes.

### CODE REVIEW

**Review by Brunoatf — COMMENTED** [review]
Boa! Só precisamos remover e ignoraro conteúdo de node_modules, acho que a sintaxe do git ignore deveria ser node_modules/

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      - db
+
+  frontend:
+    build:
+      context: ./frontend
+      dockerfile: Dockerfile
+    ports:
+      - "3000:3000"
+    environment:
+      - NODE_ENV=production
```
Comment (Brunoatf) [inline docker-compose.yml:20]: onde essa env é usada?

**Review by Brunoatf — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (gmikio) [reply docker-compose.yml:20]: A variável "NODE_ENV" é uma variável interna utilizada pelo Node.js e React.js a qual define se a build será de Desenvolvimento ou Produção. Caso seja de Produção, diversos pacotes internos são ajustados para tal, otimizando a forma de renderização do React, e também evitando instalar os pacotes designados como "devDependencies", como Logs etc.

File: frontend/.eslintrc.json [hunk 2]
```diff
@@ -1,3 +1,36 @@
...
+  "env": {
+    "browser": true,
+    "node": true,
+    "es6": true
+  },
+  "rules": {
+    "no-console": "warn",
+    "react/prop-types": "off",
+    // "quotes": ["error", "single"],
+    "no-unused-vars": [
```
[the remark this answers is not in the document]
Comment (gmikio) [reply frontend/.eslintrc.json:19]: Resolvido o problema de "no_unused_vars" no authStore.js

**Review by Brunoatf — APPROVED** [review]

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+services:
+  backend:
+    build:
+      context: ./backend
+      dockerfile: Dockerfile
+    ports:
+      - "8000:8000"
+    environment:
```
[the remark this answers is not in the document]
Comment (gmikio) [reply docker-compose.yml]: Docker-compose removed. At least for now.

**Pull request merged**